r/R/util.R

```diff
@@ -251,3 +251,25 @@ augment_io_error_msg <- function(e, call, schema = NULL, format = NULL) {
   handle_augmented_field_misuse(msg, call)
   abort(msg, call = call)
 }
+
+simulate_data_frame <- function(schema) {
+
```

Comment:
Add a comment explaining why we need this function.

And do we need to export this? I think I saw some discussion between @paleolimbot and @krlmlr about needing this function in DBI or adbc.

Reply:
In DBI, we need to infer the SQL data types from a RecordBatchReader, even if the DBI backend is not aware of Arrow (yet).

I wonder if there should be a way to convert a schema to a zero-row Arrow table. One way to do this is via schema -> zero-row table -> data frame. For DBI, schema -> data frame is sufficient, but perhaps schema -> zero-row table is easier to implemented in C++, and the table -> data frame operation is already efficient enough for zero-row tables.